### PR TITLE
feat: define project structure

### LINKED ISSUE #30: feat: define project structure
### What is the problem you're trying to solve?

Define basic structure for the project

### Task list

- [x] Create folder for protos
- [x] Create folder for each team
- [x] Init lib for protos
- [x] Import lib in all projects, except for the CLI

### Anything else ?

_No response_

### PR DESCRIPTION
**Pull request by lucido-simon**
## Related issues :
Close #30 

## Tasks lists :

- [x] Create folder for protos
- [x] Create folder for each team
- [x] Init lib for protos
- [x] Import lib in all projects, except for the CLI

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "cli"
```
[the remark this answers is not in the document]
Comment (RedbeanGit) [reply cli/Cargo.toml:2]: It seems to me more appropriate indeed

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[workspace]
+
+members = [
+    "controller",
+    "agent",
+    "networking",
```
Comment (RedbeanGit) [inline Cargo.toml:6]: Can you rename `networking` to `network` please. It's to prevent long name in the future

File: networking/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "networking"
```
Comment (RedbeanGit) [inline networking/Cargo.toml:2]: Same here

**Review by RedbeanGit — APPROVED** [review]
Seems good to me

File: cli/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucido-simon) [reply cli/Cargo.toml]: done !

**Review by iverly — DISMISSED** [review]

File: Cargo.toml (hunk 2 above)
Comment (lucido-simon) [reply Cargo.toml]: done !

**Review by RedbeanGit — APPROVED** [review]

File: networking/Cargo.toml (hunk 3 above)
Comment (lucido-simon) [reply networking/Cargo.toml]: done too !

**Review by NilsPonsard — APPROVED** [review]

**Review by Alexis-Bernard — APPROVED** [review]

**Review by WoodenMaiden — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
```
Comment (Roxxas96) [inline .gitignore:6]: You should remove `Cargo.lock`

**Review by Roxxas96 — APPROVED** [review]

**Review by Alexis-Bernard — APPROVED** [review]

**Pull request merged**